src/telegraf.ts

```diff
@@ -116,17 +117,30 @@ export class Telegraf<
 
   get webhookReply() {
     return this.telegram.webhookReply
-  } /* eslint brace-style: 0 */
+  }
 
   catch(handler: (err: any, ctx: TContext) => void) {
     this.handleError = handler
     return this
   }
 
   webhookCallback(path = '/') {
+    let botInfoCall: Promise<void> | undefined
     return generateCallback(
       path,
-      (update: tt.Update, res: any) => this.handleUpdate(update, res),
+      async (update: tt.Update, res: http.ServerResponse) => {
+        if (!this.botInfo) {
+          await (botInfoCall ??= (async () => {
+            try {
+              this.botInfo ??= await this.telegram.getMe()
+            } catch (err) {
+              debug('Could not initialize `botInfo`', err)
+              throw err
+            }
+          })())
+        }
```

Comment:
Could be significantly simplified if you're willing to drop the `debug` call:

```suggestion
        this.botInfo ??= await (botInfoCall ??= this.telegram.getMe())
```

Reply:
Yeah I think the error message should be enough. Thanks for the hint :+1: